src/main/java/seedu/address/logic/commands/AddMcqCommand.java

```diff
@@ -1,61 +1,65 @@
 package seedu.address.logic.commands;
 
 import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.commands.AddCommand.MESSAGE_DUPLICATE_QUESTION;
 import static seedu.address.logic.parser.CliSyntax.PREFIX_ANSWER;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_IMPORTANCE;
 import static seedu.address.logic.parser.CliSyntax.PREFIX_OPTION;
 import static seedu.address.logic.parser.CliSyntax.PREFIX_QUESTION;
 
-import java.util.Arrays;
-
+import seedu.address.logic.commands.exceptions.CommandException;
 import seedu.address.model.Model;
+import seedu.address.model.question.Question;
 
 /**
  * Adds an MCQ question
  */
 public class AddMcqCommand extends Command {
 
-    public static final String COMMAND_WORD = "/mcq";
-    public static final String MESSAGE_SUCCESS = "New question added: question: %s options: %s answer: %s";
+    public static final String COMMAND_WORD = "mcq";
+    public static final String MESSAGE_SUCCESS = "New question added: question: %s";
     public static final String MESSAGE_USAGE = COMMAND_WORD + ": Adds a question to SmartNUS. "
             + "Parameters: "
             + PREFIX_QUESTION + "QUESTION "
             + PREFIX_OPTION + "OPTION "
             + PREFIX_OPTION + "OPTION "
             + PREFIX_OPTION + "OPTION "
             + PREFIX_ANSWER + "ANSWER "
+            + PREFIX_IMPORTANCE + "IMPORTANCE "
             + "Example: " + COMMAND_WORD + " "
             + PREFIX_QUESTION + "What is 1+1? "
             + PREFIX_OPTION + "3 "
             + PREFIX_OPTION + "4 "
             + PREFIX_OPTION + "5 "
-            + PREFIX_ANSWER + "2 ";
+            + PREFIX_ANSWER + "2 "
+            + PREFIX_IMPORTANCE + "1 ";
 
-    // Temp attributes before implementation of Question Object
-    private final String question;
-    private final String[] options;
-    private final String answer;
+    private final Question toAdd;
 
     /**
      * Temporary constructor
      */
-    public AddMcqCommand(String question, String[] options, String answer) {
-        this.question = question;
-        this.options = options;
-        this.answer = answer;
+    public AddMcqCommand(Question question) {
+        this.toAdd = question;
     }
 
     @Override
-    public CommandResult execute(Model model) {
+    public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
-        return new CommandResult(String.format(MESSAGE_SUCCESS, question, Arrays.toString(options), answer));
+
+        if (model.hasQuestion(toAdd)) {
+            throw new CommandException(MESSAGE_DUPLICATE_QUESTION);
+        }
+
+        model.addQuestion(toAdd);
```

Comment:
I think we can add a test for this line similar to this test: execute_duplicateQuestion_throwsCommandException().

Reply:
Added tests for AddMcqCommand